### ISSUE TITLE
[Objects] Inheritence in bricks bug

### ISSUE THREAD

**Issue by jremmurd** [pimcore/pimcore#6455]
# Bug Report
Verified in Pimcore v6.6.2

### Use Case:

I want set a field on an object's child brick field. The parent has a brick with a field A which has a value, the children sets a value to another field B of that brick. Now the children brick should inherit the value of field A and should have a value set for field B, but field A is not empty an thus the value is not inherited, but the value of the parent is set explicitly to the child object.

The behaviour is the same if  `$variant` was just created as well as if it already existed.

### Dummy code (to reproduce)
```php
// all listeners disabled
// inheritence enabled

$product = Product::getById(123);

$variant = new Product();
$variant->setType(AbstractObject::OBJECT_TYPE_VARIANT); // also does not work if its a regular object
$variant->setParent($product);
$variant->setPublished(true);
$variant->setKey(time());

if ($brickContainer = $variant ->getMyBrickField()) {
    if (!($brick = $brickContainer->getMyBrick())) {
        $brick = new MyBrick($variant );
        $brickContainer->setMyBrick($brick);
    }
}

$brick->setPrice(100);

$variant ->save();
```

![image](https://user-images.githubusercontent.com/9052094/82886399-c3d0f580-9f46-11ea-85cb-9cf69cf73840.png)

![image](https://user-images.githubusercontent.com/9052094/82886734-39d55c80-9f47-11ea-9004-2f88d173797d.png)

**Comment by dvesh3** [comment]
@jremmurd are you using default value for that field? might be related https://github.com/pimcore/pimcore/issues/6376

**Comment by jremmurd** [comment]
> @jremmurd are you using default value for that field? might be related #6376

No, no default value is set. Just checked again it also occures for simple numeric inputs.

**Comment by dvesh3** [comment]
fixed in 6.6.3 https://github.com/pimcore/pimcore/pull/6342
